### PR TITLE
Feature/fcla 61 athletic preparation prototype #2

### PR DESCRIPTION
**Pull request by nlasagni**
This is the super-duper first prototype of the Athletic Preparation Bounded Context domain.
It will be refactored (maybe a lot) in the next versions, with the following reasoning about the domain.

### CODE REVIEW

File: src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/adapter/repository/InMemoryAthleticPreparationRepository.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package it.unibo.lss.fcla.athleticpreparation.adapter.repository
+
+import it.unibo.lss.fcla.athleticpreparation.domain.model.AthleticPreparation
+import it.unibo.lss.fcla.athleticpreparation.domain.model.AthleticPreparationId
+import it.unibo.lss.fcla.athleticpreparation.domain.model.PeriodOfPreparation
+import it.unibo.lss.fcla.athleticpreparation.usecase.port.AthleticPreparationRepository
+
+/**
+ * @author Nicola Lasagni on 07/03/2021.
+ */
```
Comment (sbraggion) [inline src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/adapter/repository/InMemoryAthleticPreparationRepository.kt:12]: missing javadoc

File: src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/usecase/CreateAthleticPreparation.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package it.unibo.lss.fcla.athleticpreparation.usecase
+
+import it.unibo.lss.fcla.athleticpreparation.domain.model.AthleticPreparation
+import it.unibo.lss.fcla.athleticpreparation.usecase.exception.AthleticPreparationWithSameIdAlreadyExists
+import it.unibo.lss.fcla.athleticpreparation.usecase.exception.AthleticPreparationWithSamePeriodAlreadyExists
+import it.unibo.lss.fcla.athleticpreparation.usecase.port.AthleticPreparationRepository
+
+/**
+ * @author Nicola Lasagni on 06/03/2021.
```
Comment (sbraggion) [inline src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/usecase/CreateAthleticPreparation.kt:9]: missing javadoc

File: src/test/kotlin/it/unibo/lss/fcla/athleticpreparation/adapter/repository/InMemoryAthleticPreparationRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import it.unibo.lss.fcla.athleticpreparation.domain.model.AthleticPreparation
+import it.unibo.lss.fcla.athleticpreparation.domain.model.AthleticPreparationId
+import it.unibo.lss.fcla.athleticpreparation.domain.model.PeriodOfPreparation
+import it.unibo.lss.fcla.athleticpreparation.usecase.port.AthleticPreparationRepository
+import org.junit.jupiter.api.Assertions
+import java.time.LocalDate
+
+/**
+ * @author Nicola Lasagni on 07/03/2021.
+ */
```
Comment (sbraggion) [inline src/test/kotlin/it/unibo/lss/fcla/athleticpreparation/adapter/repository/InMemoryAthleticPreparationRepositoryTest.kt:15]: missing javadoc

File: src/test/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/AthleticPreparationTest.kt [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import it.unibo.lss.fcla.athleticpreparation.domain.exception.AthleticPreparationMustHaveMember
+import it.unibo.lss.fcla.athleticpreparation.domain.exception.TrainingPlanMustNotOverlap
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import java.time.LocalDate
+
+/**
+ * @author Nicola Lasagni on 24/02/2021.
+ */
```
Comment (sbraggion) [inline src/test/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/AthleticPreparationTest.kt:17]: missing javadoc

**Review by sbraggion — DISMISSED** [review]
improve documentation in next release

File: src/test/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/ExerciseTest.kt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package it.unibo.lss.fcla.athleticpreparation.domain.model
+
+import io.kotest.core.spec.style.FreeSpec
+
+/**
+ * @author Nicola Lasagni on 04/03/2021.
+ */
+class ExerciseTest : FreeSpec({
```
Comment (Giordo94) [inline src/test/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/ExerciseTest.kt]: Add at least a test or remove this class

File: src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/AthleticPreparation.kt [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+ * decides that it is completed.
+ *
+ * @property athleticTrainerId The id reference of the athletic trainer who is preparing the athletic preparation.
+ * @property memberId The id reference of the member for whom the athletic preparation is being prepared.
+ * @property periodOfPreparation The period of athletic preparation. See [PeriodOfPreparation].
+ *
+ * @author Nicola Lasagni on 22/02/2021.
+ */
+class AthleticPreparation(
+    private val athleticTrainerId: String,
```
Comment (Giordo94) [inline src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/AthleticPreparation.kt]: Should be consistent with other ID, replacing string with a wrapper class

File: src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/AthleticPreparation.kt [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    private val memberId: String,
```
Comment (Giordo94) [inline src/main/kotlin/it/unibo/lss/fcla/athleticpreparation/domain/model/AthleticPreparation.kt]: Should be consistent with other ID, replacing string with a wrapper class

**Review by Giordo94 — CHANGES_REQUESTED** [review]
Some small changes required

**Review by Giordo94 — APPROVED** [review]

**Pull request merged**